archinstall/lib/interactions/system_conf.py

```diff
@@ -113,6 +110,32 @@ def ask_for_swap(preset: bool = True) -> bool:
 		case ResultType.Skip:
 			return preset
 		case ResultType.Selection:
-			return result.item() == MenuItem.yes()
+			enabled = result.item() == MenuItem.yes()
+			if not enabled:
+				return ZramConfiguration(enabled=False)
+
+			# Ask for compression algorithm
+			algo_items = [MenuItem(a.value, value=a) for a in ZramAlgorithm]
+			algo_group = MenuItemGroup(algo_items, sort_items=False)
+			algo_group.set_default_by_value(ZramAlgorithm.ZSTD)
+			algo_group.set_focus_by_value(preset.algorithm)
+
+			algo_result = SelectMenu[ZramAlgorithm](
+				algo_group,
+				header=tr('Select zram compression algorithm:') + '\n',
+				alignment=Alignment.CENTER,
+				allow_skip=True,
+			).run()
+
+			algo: ZramAlgorithm
+			match algo_result.type_:
+				case ResultType.Skip:
+					algo = preset.algorithm
+				case ResultType.Selection:
+					algo = algo_result.get_value()
+				case ResultType.Reset:
```

Comment:
Reset is not enabled in the `SelectMenu` (`allow_reset` is default to false) so this is unnecessary